scrape/scrape.go

```diff
@@ -1184,7 +1185,7 @@ loop:
 		return total, added, err
 	}
 
-	sl.cache.iterDone()
+	sl.cache.iterDone(len(b) > 0)
```

Comment:
this deserves a comment explaining why this condition makes it right to drop cache, also extracting it here into a variable that makes it obvious already what this condition does would be helpful as well

Reply:
Done